heron/schedulers/src/java/com/twitter/heron/scheduler/kubernetes/KubernetesScheduler.java

```diff
@@ -253,4 +374,52 @@ protected String getFetchCommand() {
 
     return command;
   }
+
+  @Override
+  public void addContainers(Set<PackingPlan.ContainerPlan> containersToAdd) {
+    // grab the base pod so we can copy and modify some stuff
+    String basePodName = Runtime.topologyName(runtime) + "-0";
+    JsonNode podConfig;
+    try {
+      podConfig = controller.getBasePod(basePodName);
+    } catch (IOException ioe) {
+      throw new TopologyRuntimeManagementException("Unable to retrieve base pod configuration from "
+          + basePodName, ioe);
+    }
+
+    // iterate over the containers we need to add and rebuild the spec based on the new plan
+    ObjectMapper mapper = new ObjectMapper();
+    int totalNewContainerCount = containersToAdd.size();
+    int deployedContainerCount = 0;
+    for (PackingPlan.ContainerPlan containerPlan : containersToAdd) {
+      String newContainer = rebuildKubernetesPodSpec(podConfig, mapper, containerPlan, 0);
+
+      // deploy this new container
+      try {
+        controller.deployContainer(newContainer);
+        deployedContainerCount++;
+        LOG.log(Level.INFO, "New container " + ++deployedContainerCount + "/"
+            + totalNewContainerCount + " deployed");
+      } catch (IOException ioe) {
+        LOG.log(Level.SEVERE, "Issue deploying a container! Previously added container count: "
+            + deployedContainerCount);
+        LOG.log(Level.INFO, "Configuration of problem container: " + newContainer);
+        throw new TopologyRuntimeManagementException("Problem adding container with id "
```

Comment:
can we remove the logging here and add any relevant info to the exception message? Instead of log info, log sever and throw exception, best to just throw the exception with a good message.